Tokens issued with incorrect scopes for Access Points (SOFTWARE-4772)

brianhlin opened the pull request:
Adding the server side checks to make sure that AP vs EE token requests have the correct scopes. Still need to work on updating the client but figured that this was enough for an initial review.

brianhlin committed:
Separate EE vs AP Topology queries (SOFTWARE-4772)

brianhlin committed:
Refactor the check for resource admin for the requested token

brianhlin committed:
Ensure that allowed lists exist to avoid NameErrors

brianhlin committed:
Ensure correct scopes for AP vs EE resources (SOFTWARE-4772)

bbockelm commented on registry/token/views.py:
```diff
@@ -24,7 +24,9 @@ def code_get():
 SOURCE_POSTFIX = "flock.opensciencegrid.org"
-ALLOWED_AUTHORIZATIONS = {"READ", "ADVERTISE_STARTD", "ADVERTISE_MASTER"}
+BASE_ALLOWED_AUTHORIZATIONS = {"READ", "ADVERTISE_MASTER"}
+AP_ALLOWED_AUTHORIZATIONS = {"ADVERTISE_SCHEDD", "DAEMON"}
```
Can we leave out `DAEMON` for now?

I don't think it's required even if it shuts up complaints in the logs.

brianhlin replied:
Done

brianhlin committed:
Drop DAEMON scope

We'll put up with annoying (but harmless) errors in the logs to avoid
the proliferation of tokens with the DAEMON scope

bbockelm reviewed (approved):
Ok, I think we're good to go.

Obviously I have not tested the functionality but the concepts look correct.

brianhlin commented:
Let's see if it works on the ITB! Client bits to come in a separate PR
